src/equations/compressible_euler_1d.jl

```diff
@@ -864,8 +864,9 @@ Compactly summarized:
     beta = sqrt(0.5f0 * (equations.gamma - 1) / equations.gamma)
 
     # Estimate the edges of the Riemann fan (with positivity conservation)
-    SsL = min(v_roe - c_roe, v_ll - beta * c_ll, 0)
-    SsR = max(v_roe + c_roe, v_rr + beta * c_rr, 0)
+    RealT = eltype(u_ll)
+    SsL = min(v_roe - c_roe, v_ll - beta * c_ll, zero(RealT))
+    SsR = max(v_roe + c_roe, v_rr + beta * c_rr, zero(RealT))
```

Comment:
By self-review. Consider that `SsL` and `SsR` may be used in other functions, potentially being called by functions such as `sqrt()` or `inv()` (has not been checked yet, but if they are integers then it is going to produce `Float64`). It is better to make changes here. How do you think?

Reply:
In this case, `v_roe` and `c_roe` etc. will already be floating point types. Thus, `min` and `max` promote the integer `0` to floating point types of the correct type.

```julia
julia> min(1.0f0, 0)
0.0f0

julia> max(1.0, 0)
1.0
```

Reply:
Sorry :( 
First time I checked the promotion of `min` `max` functions and later I forgot these things.